### ISSUE TITLE
neutronless.loadbalancer.test_featureoff_create_delete_basic_lb

### ISSUE THREAD

**Issue by ssorenso** [F5Networks/f5-openstack-agent#885]
* Title: A short but descriptive summary of the issue, whether it be a bug or enhancement.
* Attachments: 
* Details: 

##### Issue level suggestion
Bug

We're expecting that the member exists, the test appears to be working, and it does not; thus, some kind of functionality failed.

##### Traceback
```
bigip = <neutronless.testlib.bigip_client.BigIpClient object at 0x7f91529e6b10>
services = [{'healthmonitors': [], 'l7policies': [], 'l7policy_rules': [], 'listeners': [], ...}, {'healthmonitors': [], 'l7polic...ules': [], 'listeners': [], ...}, {'healthmonitors': [], 'l7policies': [], 'l7policy_rules': [], 'listeners': [], ...}]
icd_config = {'AGENT': '<oslo_config.cfg.GroupAttr object at 0x3b137d0>', 'OVS': '<oslo_config.cfg.GroupAttr object at 0x3ae9410>', 'advertise_mtu': False, 'advertised_tunnel_types': ['vxlan'], ...}
icontrol_driver = <f5_openstack_agent.lbaasv2.drivers.bigip.icontrol_driver.iControlDriver object at 0x7f9152c557d0>

    def test_featureoff_create_delete_basic_lb(bigip, services, icd_config, icontrol_driver):

        service_iter = iter(services)
        service = service_iter.next()
        lb_reader = LoadbalancerReader(service)
        env_prefix = icd_config['environment_prefix']
        fake_rpc = icontrol_driver.plugin_rpc
        hostname = bigip.get_device_name()
        icd_config['f5_network_segment_physical_network'] = None

        folder = '%s_%s' % (env_prefix, lb_reader.tenant_id())

        # Make sure we are starting clean.
        assert not bigip.folder_exists(folder)

        # Create the loadbalancer
        retval = icontrol_driver._common_service_handler(service)
        assert not retval

        # Assert that update loadbalancer status was not called
        assert fake_rpc.get_call_count('update_loadbalancer_status') == 1

        # Assert folder created
>       assert bigip.folder_exists(folder)
E       AssertionError: assert False
E        +  where False = <bound method BigIpClient.folder_exists of <neutronless.testlib.bigip_client.BigIpClient object at 0x7f91529e6b10>>('TEST_d758799e02fa4b9ab79e560906df7d7f')
E        +    where <bound method BigIpClient.folder_exists of <neutronless.testlib.bigip_client.BigIpClient object at 0x7f91529e6b10>> = <neutronless.testlib.bigip_client.BigIpClient object at 0x7f91529e6b10>.folder_exists

../neutronless/loadbalancer/test_loadbalancer_opflex.py:168: AssertionError
```

##### Tests impacted:
* `openstack_agent_mitaka_12.1.2-overcloud.12.consoleText`
  * Type: failures
  * Test: `test_featureoff_create_delete_basic_lb`
  * Sum: `md5sum:490f70513f2dc19438c4908ffc219a89`

#### OpenStack Release
Mitaka

#### Agent Version
Mitaka - `e46a3c98070772ff27da829e91c8f5decb822056`

#### Operating System
Nightly

**Comment by ssorenso** [comment]
##### Traceback:
```
bigip = <neutronless.testlib.bigip_client.BigIpClient object at 0x7f6c0518fe90>
services = [{'healthmonitors': [], 'l7policies': [], 'l7policy_rules': [], 'listeners': [], ...}, {'healthmonitors': [], 'l7polic...it': -1, 'default_pool_id': '75026460-addc-4edc-ba90-50bed675b54a', 'default_tls_container_id': None, ...}], ...}, ...]
icd_config = {'AGENT': '<oslo_config.cfg.GroupAttr object at 0x3b137d0>', 'OVS': '<oslo_config.cfg.GroupAttr object at 0x3ae9410>', 'advertise_mtu': False, 'advertised_tunnel_types': ['vxlan'], ...}
icontrol_driver = <f5_openstack_agent.lbaasv2.drivers.bigip.icontrol_driver.iControlDriver object at 0x7f6c04fa3bd0>

    def test_pool_lb_change_ratio(bigip, services, icd_config, icontrol_driver):
        env_prefix = icd_config['environment_prefix']
        service_iter = iter(services)
        validator = ResourceValidator(bigip, env_prefix)

        # create lb
        service = service_iter.next()
        lb_reader = LoadbalancerReader(service)
        folder = '{0}_{1}'.format(env_prefix, lb_reader.tenant_id())
        icontrol_driver._common_service_handler(service)
>       assert bigip.folder_exists(folder)
E       AssertionError: assert False
E        +  where False = <bound method BigIpClient.folder_exists of <neutronless.testlib.bigip_client.BigIpClient object at 0x7f6c0518fe90>>('TEST_980e3f914f3e40359c3c2d9470fb2e8a')
E        +    where <bound method BigIpClient.folder_exists of <neutronless.testlib.bigip_client.BigIpClient object at 0x7f6c0518fe90>> = <neutronless.testlib.bigip_client.BigIpClient object at 0x7f6c0518fe90>.folder_exists

../neutronless/pool/test_pool_lb_method_change.py:53: AssertionError
```
* `openstack_agent_mitaka_12.1.2-overcloud.12.consoleText`
  * Type: failures
  * Test: `test_pool_lb_change_ratio`
  * Sum: `md5sum:5866396773725662dde832797c8f2ac6`

**Comment by ssorenso** [comment]
Another instance:
```
request = <FixtureRequest for <Function 'test_lbaas_service_one_opflex_member'>>
bigip = <neutronless.testlib.bigip_client.BigIpClient object at 0x7fdf77400f90>
services = [{'healthmonitors': [], 'l7policies': [], 'l7policy_rules': [], 'listeners': [{'admin_state_up': True, 'connection_lim...n_limit': -1, 'default_pool_id': 'b67d6530-8a82-43d4-a1dc-3418f3e62411', 'default_tls_container_id': None, ...}], ...}]
icd_config = {'AGENT': '<oslo_config.cfg.GroupAttr object at 0x3b137d0>', 'OVS': '<oslo_config.cfg.GroupAttr object at 0x3ae9410>', 'advertise_mtu': False, 'advertised_tunnel_types': ['vxlan'], ...}
icontrol_driver = <f5_openstack_agent.lbaasv2.drivers.bigip.icontrol_driver.iControlDriver object at 0x7fdf77461a50>

    def test_lbaas_service_one_opflex_member(request,
                                        bigip,
                                        services,
                                        icd_config,
                                        icontrol_driver):
        env_prefix = icd_config['environment_prefix']
        service_iter = iter(services)
        service = service_iter.next()
        lb_reader = LoadbalancerReader(service)
        fake_rpc = icontrol_driver.plugin_rpc
        folder = '{0}_{1}'.format(env_prefix, lb_reader.tenant_id())

        lb_pending = icontrol_driver._common_service_handler(service)
>       assert bigip.folder_exists(folder)
E       AssertionError: assert False
E        +  where False = <bound method BigIpClient.folder_exists of <neutronless.testlib.bigip_client.BigIpClient object at 0x7fdf77400f90>>('TEST_0e79c89d48824e25bda3366280f801a5')
E        +    where <bound method BigIpClient.folder_exists of <neutronless.testlib.bigip_client.BigIpClient object at 0x7fdf77400f90>> = <neutronless.testlib.bigip_client.BigIpClient object at 0x7fdf77400f90>.folder_exists

../neutronless/member/test_single_member_opflex.py:57: AssertionError
```
* `openstack_agent_mitaka_12.1.2-overcloud.12.consoleText`
  * Type: failures
  * Test: `test_lbaas_service_one_opflex_member`
  * Sum: `md5sum:6b2ee7025e8b10cba72b1a7cc869a1c6`

**Comment by ssorenso** [comment]
```
bigip = <neutronless.testlib.bigip_client.BigIpClient object at 0x7f61d7be1b10>
services = [{'healthmonitors': [], 'listeners': [], 'loadbalancer': {'admin_state_up': True, 'description': 'create lb', 'gre_vte...ol session persistence', 'gre_vteps': [], 'id': '54e91746-e33f-4e0a-b04a-9ecba797f752', ...}, 'members': [], ...}, ...]
icd_config = {'AGENT': '<oslo_config.cfg.GroupAttr object at 0x3b137d0>', 'OVS': '<oslo_config.cfg.GroupAttr object at 0x3ae9410>', 'advertise_mtu': False, 'advertised_tunnel_types': ['vxlan'], ...}
icontrol_driver = <f5_openstack_agent.lbaasv2.drivers.bigip.icontrol_driver.iControlDriver object at 0x7f61d7ab9310>

    def test_single_pool_tcp_vs(bigip, services, icd_config, icontrol_driver):
        env_prefix = icd_config['environment_prefix']
        service_iter = iter(services)
        validator = ResourceValidator(bigip, env_prefix)

        # create loadbalancer
        service = service_iter.next()
        lb_reader = LoadbalancerReader(service)
        folder = '{0}_{1}'.format(env_prefix, lb_reader.tenant_id())
        icontrol_driver._common_service_handler(service)
>       assert bigip.folder_exists(folder)
E       AssertionError: assert False
E        +  where False = <bound method BigIpClient.folder_exists of <neutronless.testlib.bigip_client.BigIpClient object at 0x7f61d7be1b10>>('TEST_26e7ccac8d54491cb1447ebc738a6803')
E        +    where <bound method BigIpClient.folder_exists of <neutronless.testlib.bigip_client.BigIpClient object at 0x7f61d7be1b10>> = <neutronless.testlib.bigip_client.BigIpClient object at 0x7f61d7be1b10>.folder_exists

../neutronless/listener/test_tcp_listener.py:51: AssertionError
```
* `openstack_agent_mitaka_12.1.2-overcloud.12.consoleText`
  * Type: failures
  * Test: `test_single_pool`
  * Sum: `md5sum:a9f96ba209a057321dc04899227e7c30`

**Comment by ssorenso** [comment]
```
bigip = <neutronless.testlib.bigip_client.BigIpClient object at 0x7f61d7be1b10>
services = [{'healthmonitors': [], 'listeners': [], 'loadbalancer': {'admin_state_up': True, 'description': 'create lb', 'gre_vte...ol session persistence', 'gre_vteps': [], 'id': '54e91746-e33f-4e0a-b04a-9ecba797f752', ...}, 'members': [], ...}, ...]
icd_config = {'AGENT': '<oslo_config.cfg.GroupAttr object at 0x3b137d0>', 'OVS': '<oslo_config.cfg.GroupAttr object at 0x3ae9410>', 'advertise_mtu': False, 'advertised_tunnel_types': ['vxlan'], ...}
icontrol_driver = <f5_openstack_agent.lbaasv2.drivers.bigip.icontrol_driver.iControlDriver object at 0x7f61d7ab9310>

    def test_single_pool_tcp_vs(bigip, services, icd_config, icontrol_driver):
        env_prefix = icd_config['environment_prefix']
        service_iter = iter(services)
        validator = ResourceValidator(bigip, env_prefix)

        # create loadbalancer
        service = service_iter.next()
        lb_reader = LoadbalancerReader(service)
        folder = '{0}_{1}'.format(env_prefix, lb_reader.tenant_id())
        icontrol_driver._common_service_handler(service)
>       assert bigip.folder_exists(folder)
E       AssertionError: assert False
E        +  where False = <bound method BigIpClient.folder_exists of <neutronless.testlib.bigip_client.BigIpClient object at 0x7f61d7be1b10>>('TEST_26e7ccac8d54491cb1447ebc738a6803')
E        +    where <bound method BigIpClient.folder_exists of <neutronless.testlib.bigip_client.BigIpClient object at 0x7f61d7be1b10>> = <neutronless.testlib.bigip_client.BigIpClient object at 0x7f61d7be1b10>.folder_exists

../neutronless/listener/test_tcp_listener.py:51: AssertionError
```
* `openstack_agent_mitaka_12.1.2-overcloud.12.consoleText`
  * Type: failures
  * Test: `test_single_pool_tcp_vs`
  * Sum: `md5sum:d1363dcb53eebb14b41568c6eb02036e`

**Comment by ssorenso** [comment]
```
bigip = <neutronless.testlib.bigip_client.BigIpClient object at 0x7f4ca8977050>
services = [{'healthmonitors': [], 'l7policies': [], 'l7policy_rules': [], 'listeners': [], ...}, {'healthmonitors': [], 'l7polic...it': -1, 'default_pool_id': 'e70cedf7-6305-4f3c-91aa-6a14ae58136c', 'default_tls_container_id': None, ...}], ...}, ...]
icd_config = {'AGENT': '<oslo_config.cfg.GroupAttr object at 0x3b137d0>', 'OVS': '<oslo_config.cfg.GroupAttr object at 0x3ae9410>', 'advertise_mtu': False, 'advertised_tunnel_types': ['vxlan'], ...}
icontrol_driver = <f5_openstack_agent.lbaasv2.drivers.bigip.icontrol_driver.iControlDriver object at 0x7f4ca87a7dd0>
esd = {'esd_demo_1': {'lbaas_cssl_profile': 'clientssl', 'lbaas_ctcp': 'tcp-mobile-optimized', 'lbaas_fallback_persist': 'so...ys_APM_ExchangeSupport_helper']}, 'esd_demo_3': {'lbaas_ctcp': 'tcp-wan-optimized', 'lbaas_stcp': 'tcp-lan-optimized'}}

    def test_esd(bigip, services, icd_config, icontrol_driver, esd):
        env_prefix = icd_config['environment_prefix']
        service_iter = iter(services)
        validator = ResourceValidator(bigip, env_prefix)

        # create loadbalancer
        service = service_iter.next()
        lb_reader = LoadbalancerReader(service)
        folder = '{0}_{1}'.format(env_prefix, lb_reader.tenant_id())
        icontrol_driver._common_service_handler(service)
>       assert bigip.folder_exists(folder)
E       AssertionError: assert False
E        +  where False = <bound method BigIpClient.folder_exists of <neutronless.testlib.bigip_client.BigIpClient object at 0x7f4ca8977050>>('TEST_ba91c2d0dee44bea8f235bed14519a8c')
E        +    where <bound method BigIpClient.folder_exists of <neutronless.testlib.bigip_client.BigIpClient object at 0x7f4ca8977050>> = <neutronless.testlib.bigip_client.BigIpClient object at 0x7f4ca8977050>.folder_exists

../neutronless/esd/test_esd.py:115: AssertionError
```
* `openstack_agent_mitaka_12.1.2-overcloud.12.consoleText`
  * Type: failures
  * Test: `test_esd`
  * Sum: `md5sum:da11a32935ddc889d9e0497be44a93ff`

**Comment by mattgreene** [comment]
Regression introduced with #882 and fixed with #894.
